src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentationTest.groovy

```diff
@@ -225,4 +230,40 @@ class DataLoaderDispatcherInstrumentationTest extends Specification {
         starWarsWiring.batchFunctionLoadCount == 2
         starWarsWiring.naiveLoadCount == 8
     }
+
+    def "can be efficient with lazily computed data loaders"() {
```

Comment:
maybe we could add a negative test, where we attempt to register a dataloader onto the default registry and expect it to blow up.